### PR TITLE
Add FlushAll api

### PR DESCRIPTION
**Pull request by bigsheeper**
related: https://github.com/milvus-io/milvus/issues/22921

### CODE REVIEW

File: pymilvus/client/grpc_handler.py [hunk 1]
```diff
@@ -1293,3 +1293,50 @@ def transfer_replica(self, source, target, collection_name, num_replica, timeout
...
+                if end - start > timeout:
+                    raise MilvusException(message=f"wait for flush all timeout, flush_all_ts: {flush_all_ts}")
+
+            if not flush_ret:
+                time.sleep(5)
+
+    @retry_on_rpc_failure()
+    def flush_all(self, timeout=None, **kwargs):
+        request = Prepare.flush_all_request()
+        future = self._stub.FlushAll.future(request, timeout=timeout)
```
Comment (longjiquan) [inline pymilvus/client/grpc_handler.py:1323]: Remove `future`.

File: pymilvus/client/grpc_handler.py [hunk 2]
```diff
@@ -1293,3 +1293,50 @@ def transfer_replica(self, source, target, collection_name, num_replica, timeout
         resp = self._stub.TransferReplica(req, wait_for_ready=True, timeout=timeout)
         if resp.error_code != 0:
             raise MilvusException(resp.error_code, resp.reason)
+
+    @retry_on_rpc_failure()
+    def get_flush_all_state(self, flush_all_ts, timeout=None, **kwargs):
+        req = Prepare.get_flush_all_state_request(flush_all_ts)
+        future = self._stub.GetFlushAllState.future(req, timeout=timeout)
```
Comment (longjiquan) [inline pymilvus/client/grpc_handler.py:1300]: Remove `future`.

File: pymilvus/orm/utility.py [hunk 3]
```diff
@@ -1036,3 +1036,25 @@ def transfer_replica(source_group, target_group, collection_name, num_replicas,
         >>> rgs = utility.transfer_replica(source, target, collection_name, num_replica)
     """
     return _get_connection(using).transfer_replica(source_group, target_group, collection_name, num_replicas, timeout)
+
+def flush_all(using="default", timeout=None):
```
Comment (longjiquan) [inline pymilvus/orm/utility.py:1040]: Add `kwargs` to asnychronous support. Also, tell the users how to use the asynchronous ability in the document comment.

Comment (bigsheeper) [reply pymilvus/orm/utility.py:1040]: done

File: pymilvus/client/grpc_handler.py [hunk 4]
```diff
@@ -1293,3 +1293,50 @@ def transfer_replica(self, source, target, collection_name, num_replica, timeout
...
+                    raise MilvusException(message=f"wait for flush all timeout, flush_all_ts: {flush_all_ts}")
+
+            if not flush_ret:
+                time.sleep(5)
+
+    @retry_on_rpc_failure()
+    def flush_all(self, timeout=None, **kwargs):
+        request = Prepare.flush_all_request()
+        future = self._stub.FlushAll.future(request, timeout=timeout)
+        flush_all_ts = future.result().flush_all_ts
```
Comment (longjiquan) [inline pymilvus/client/grpc_handler.py:1323]: Please don't call `future.result()` twice.

File: pymilvus/orm/utility.py [hunk 5]
```diff
@@ -1036,3 +1036,29 @@ def transfer_replica(source_group, target_group, collection_name, num_replicas,
...
+        >>> from pymilvus import connections, Collection, FieldSchema, CollectionSchema, DataType, utility
+        >>> connections.connect()
+        >>> fields = [
+        ...     FieldSchema("film_id", DataType.INT64, is_primary=True),
+        ...     FieldSchema("films", dtype=DataType.FLOAT_VECTOR, dim=128)
+        ... ]
+        >>> schema = CollectionSchema(fields=fields)
+        >>> collection = Collection(name="test_collection_flush", schema=schema)
+        >>> collection.insert([[1, 2], [[1.0, 2.0], [3.0, 4.0]]])
+        >>> utility.flush_all()
```
Comment (longjiquan) [inline pymilvus/orm/utility.py:1061]: Please also add example for the asynchronous feature.

Comment (longjiquan) [reply pymilvus/orm/utility.py:1061]: And test it locally.

File: pymilvus/client/grpc_handler.py [hunk 6]
```diff
@@ -1293,3 +1293,50 @@ def transfer_replica(self, source, target, collection_name, num_replica, timeout
         resp = self._stub.TransferReplica(req, wait_for_ready=True, timeout=timeout)
         if resp.error_code != 0:
             raise MilvusException(resp.error_code, resp.reason)
+
+    @retry_on_rpc_failure()
+    def get_flush_all_state(self, flush_all_ts, timeout=None, **kwargs):
```
Comment (longjiquan) [inline pymilvus/client/grpc_handler.py:1298]: Add ut for these all newly apis.

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Pull request merged**